pkgs/applications/blockchains/solana/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, rustPlatform
+, IOKit
+, Security
+, AppKit
+, openssl
+, pkg-config
+, rustfmt
+, udev
+, zlib
+, llvm
+, llvmPackages
+, rocksdb
+, protobuf
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "solana-testnet-cli";
+  version = "1.9.2";
+
+  src = fetchFromGitHub {
+    owner = "solana-labs";
+    repo = "solana";
+    rev = "v${version}";
+    sha256 = "sha256-wrv35vBohLztMZPb6gfZdCaXcjj/Y7vnQqINaI6dBM4=";
+  };
+
+  cargoSha256 = "sha256-A5uVa+cRmrkVyw7MFH4QAr0VIFi18wcc2VPFvQyT9EM=";
+
+  buildAndTestSubdir = "cli";
+
+  nativeBuildInputs = lib.optionals stdenv.isLinux [ protobuf ];
+  buildInputs = lib.optionals stdenv.isLinux [ pkg-config udev zlib ] ++ lib.optionals stdenv.isDarwin [ IOKit Security AppKit ];
+
+  # check phase fails
+  # on darwin with missing framework System. This framework is not available in nixpkgs
+  # on linux with some librocksdb-sys compilation error
+  doCheck = false;
+  checkInputs = [ pkg-config rustfmt ];
```

Comment:
This and all of the settings below can be removed now that `doCheck = false`, right? I think it’s better to omit them for now since they don’t do anything, than to keep them where they may confuse others about their purpose.

Reply:
I've commented out stuff only needed for the checkPhase. I would like to keep it for anybody who tries to enable the checkphase.
I've removed the uneeded arguments from inputs though.